uat/testing-features/src/main/resources/greengrass/features/ggmq-1.feature

```diff
@@ -0,0 +1,32 @@
+@GGAD
+Feature: GGMQ-1
+
+  As a developer, I can configure my client agents to connect and use MQTT
+
+  Background:
+    Given my device is registered as a Thing
+    And my device is running Greengrass
+    And my MQTT Client Control is running
+
+  Scenario: GGMQ-1-T1: As a customer, I can connect, subscribe and publish using client application to MQTT topic
+    When I create a Greengrass deployment with components
+      | aws.greengrass.clientdevices.Auth        | LATEST                                                        |
+      | aws.greengrass.clientdevices.mqtt.EMQX   | LATEST                                                        |
+      | aws.greengrass.clientdevices.IPDetector  | LATEST                                                        |
+      | aws.greengrass.client.Mqtt5JavaSdkClient | classpath:/greengrass/components/recipes/client_java_sdk.yaml |
+    Then the Greengrass deployment is COMPLETED on the device after 300 seconds
+    And agent "agent1" is connected to MQTT Client Control
+    When I create client device "test" with the following policy
+      | operation | resource     |
+      | connect   | "*"          |
+      | subscribe | "iot_data_0" |
+      | publish   | "iot_data_0" |
+    And I associate "test" with ggc
+    And I connect device "test" on "agent1" to "default_broker" as "agent1-test-default_broker"
```

Comment:
What is "default_broker"? Where is that coming from?

Reply:
Since we have not implemented building blocks yet, this is only a placeholder.
I think we should have a test method with ability to specify which broker to use. Also we should have an overloaded version of this method with some defaults configured.

Reply:
That's the point of this pull request, I want us to have the test nearly exactly how I want it to look; then you go and implement the logic.

Reply:
Fixed ced07c0b79ca159d1f64195e44e23cdb61bbb7cd
I've added step to discover a broker using IPDetector. We have the assumption that we'll need some additional steps with direct configuration. 
Also let me clarify, Is it ok to run discovery once on the test application? Or you'd like us to implement discovery in test clients code?